legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -212,25 +210,39 @@ def get_company_details_revision(filing_id, legal_entity_id) -> dict:
         return company_profile_json
 
     @staticmethod
-    def get_business_revision(filing, legal_entity) -> dict:
+    def get_business_revision(filing, business) -> dict:
         """Consolidates the LegalEntity info as of a particular filing."""
-        le_revision = VersionedBusinessDetailsService.get_business_revision_obj(filing, legal_entity.id)
-        return VersionedBusinessDetailsService.business_revision_json(le_revision, legal_entity.json())
+        le_revision = VersionedBusinessDetailsService.get_business_revision_obj(filing, business)
+        return VersionedBusinessDetailsService.business_revision_json(le_revision, business.json())
 
     @staticmethod
-    def get_business_revision_obj(filing, legal_entity_id) -> LegalEntity:
+    def get_business_revision_obj(filing, business: any) -> any:
         """Return version object associated with the given filing for a LegalEntity."""
-        le_revision = LegalEntity.find_by_internal_id(legal_entity_id)
-
-        # The history table has the old revisions, not the current one.
-        if le_revision.change_filing_id != filing.id:
-            legal_entity_version = history_cls(LegalEntity)
-            le_revision = (
-                db.session.query(legal_entity_version)
-                .filter(legal_entity_version.change_filing_id == filing.id)
-                .filter(legal_entity_version.id == legal_entity_id)
-                .first()
-            )
+
+        le_revision = business
+
+        # TODO: ideally we make this one query with dynamic filters etc
+        if business.is_legal_entity:
+            # The history table has the old revisions, not the current one.
+            if business.change_filing_id and business.change_filing_id != filing.id:
+                legal_entity_version = history_cls(LegalEntity)
+                le_revision = (
+                    db.session.query(legal_entity_version)
+                    .filter(legal_entity_version.change_filing_id == filing.id)
+                    .filter(legal_entity_version.id == business.id)
+                    .first()
+                )
+        else:
+            # The history table has the old revisions, not the current one.
+            if business.change_filing_id and business.change_filing_id != filing.id:
+                alternate_name_version = history_cls(AlternateName)
+                le_revision = (
+                    db.session.query(alternate_name_version)
+                    .filter(alternate_name_version.change_filing_id == filing.id)
+                    .filter(alternate_name_version.id == business.id)
+                    .first()
+                )
```

Comment:
This returns `None` for business FM1028594 (id: 1147) and filing 142739. Checking the database confirms there is no `alternate_names_history` entry for it. Is this an issue with my db dump or do we need to handle possible `NoneType`s getting returned?

Reply:
It could be a data issue as I have not got around to populate the history entries properly for `alternate_names_history` yet.  But the scenario of it being None is possible too I believe.